crates/rbuilder/src/provider/mod.rs

```diff
@@ -51,3 +58,75 @@ pub trait RootHasher: std::fmt::Debug + Send + Sync {
     /// State root for changes outcome on top of parent block.
     fn state_root(&self, outcome: &ExecutionOutcome) -> Result<B256, RootHashError>;
 }
+
+/// All supported state provider factories
```

Comment:
I think we could add issue to use Box<StateProviderFactory> so we could simplify it later

Reply:
Not sure I follow,  how Boxing would help/simplify this one?